ui/puzzle/src/interfaces.ts

```diff
@@ -96,7 +98,7 @@ export interface Vm {
   pov: Color;
   mode: 'play' | 'view' | 'try';
   round?: PuzzleRound;
-  next: Deferred<PuzzleData | undefined>;
+  next: Deferred<PuzzleData | ReplayEnd>;
```

Comment:
that looks better imo

Reply:
Ah I see, we can store the data in the promise instead of relying on it existing elsewhere. This is a good pattern for me to keep in mind.